core/src/tile/tileWorker.cpp

```diff
@@ -30,11 +33,17 @@ TileWorker::~TileWorker(){
     }
 }
 
-void TileWorker::run() {
+void disposeBuilder(std::unique_ptr<TileBuilder> _builder) {
+    if (_builder) {
+        Tangram::runOnMainLoop([b = std::shared_ptr<TileBuilder>(std::move(_builder))](){});
```

Comment:
Let me make sure I understand this bit: The lambda here will create a `shared_ptr` that takes ownership of `_builder` and then immediately goes out of scope? I assume there's a reason for doing this instead of just resetting `_builder` - a comment here might be nice.